AX: Rename dbg -> debugDescription

hoffmanjoshua opened the pull request:
#### e81e4640901da0a2e708a8a55bf2d6875d805474
<pre>
AX: Rename dbg -&gt; debugDescription
<a href="https://bugs.webkit.org/show_bug.cgi?id=296588">https://bugs.webkit.org/show_bug.cgi?id=296588</a>
<a href="https://rdar.apple.com/156944439">rdar://156944439</a>

Reviewed by Tyler Wilcock.

Rename AXCoreObject::dbg to debugDescription, which gives it parity to the
rest of the WebKit project.

I&apos;ve also added an override to WebAccessibilityObjectWrapperBase for NSObject&apos;s
description and debugDescription methods.

* Source/WebCore/accessibility/AXCoreObject.h:
(WebCore::AXCoreObject::debugDescription const):
(WebCore::AXCoreObject::dbg const): Deleted.
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::debugDescriptionInternal const):
(WebCore::AccessibilityObject::dbgInternal const): Deleted.
* Source/WebCore/accessibility/AccessibilityObject.h:
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.cpp:
(WebCore::AXIsolatedObject::debugDescriptionInternal const):
(WebCore::AXIsolatedObject::dbgInternal const): Deleted.
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.h:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
(-[WebAccessibilityObjectWrapperBase description]):
(-[WebAccessibilityObjectWrapperBase debugDescription]):

Canonical link: <a href="https://commits.webkit.org/297981@main">https://commits.webkit.org/297981@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/18aa4a2ccde45d3b183221c471257bcf4cf90f32

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/113750 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33449 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23900 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119912 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/64577 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/115639 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/34064 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/42016 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86458 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/41567 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116698 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27168 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102175 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66819 "Found 1 new API test failure: /WPE/TestCookieManager:/webkit/WebKitCookieManager/persistent-storage (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26387 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20304 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63635 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96538 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20407 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/123148 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40744 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30375 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95302 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/41133 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98384 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95066 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24257 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40202 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18011 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/36951 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40626 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/46133 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40264 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43564 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/42039 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

twilco commented on Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
```diff
@@ -373,6 +373,13 @@ - (BOOL)_prepareAccessibilityCall
 }
 #endif
 
+- (NSString *)debugDescription
```
Why debugDescription and not description? I think the latter is what is called by default when you `po` the object in LLDB, or log like this:

```cpp
os_log_error(OS_LOG_DEFAULT, "foo %@", myWrapper);
```

So having a nicer default description through these means would be nice.

hoffmanjoshua replied:
For the logging case you have, that will call `description` (I can make `description` return the same thing). But, when you do `po` with LLDB, that should call `debugDescription`: https://developer.apple.com/documentation/objectivec/nsobjectprotocol/debugdescription?language=objc.

We can make them both do the same thing, however,

twilco replied:
Yeah, feels like it would be really nice to have this information in both contexts!

twilco commented on Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
```diff
@@ -373,6 +373,13 @@ - (BOOL)_prepareAccessibilityCall
+- (NSString *)debugDescription
+{
+    if (RefPtr<AXCoreObject> backingObject = self.axBackingObject)
+        return backingObject->debugDescription().createNSString().autorelease();
```
Is it confusing that we're just logging the backing object, making this debug description indistinguishable from the core object itself? Maybe we log: `wrapper %p { object %s }`?

twilco commented on Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
```diff
@@ -373,6 +373,13 @@ - (BOOL)_prepareAccessibilityCall
+{
+    if (RefPtr<AXCoreObject> backingObject = self.axBackingObject)
+        return backingObject->debugDescription().createNSString().autorelease();
+    return [super debugDescription];
```
Maybe nice to add that the backing object was null, e.g. the returned string could be `[super debugDescription] + (null backing object)`

hoffmanjoshua committed:
AX: Rename dbg -> debugDescription
https://bugs.webkit.org/show_bug.cgi?id=296588
rdar://156944439

Reviewed by Tyler Wilcock.

Rename AXCoreObject::dbg to debugDescription, which gives it parity to the
rest of the WebKit project.

I've also added an override to WebAccessibilityObjectWrapperBase for NSObject's
description and debugDescription methods.

* Source/WebCore/accessibility/AXCoreObject.h:
(WebCore::AXCoreObject::debugDescription const):
(WebCore::AXCoreObject::dbg const): Deleted.
* Source/WebCore/accessibility/AccessibilityObject.cpp:
(WebCore::AccessibilityObject::debugDescriptionInternal const):
(WebCore::AccessibilityObject::dbgInternal const): Deleted.
* Source/WebCore/accessibility/AccessibilityObject.h:
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.cpp:
(WebCore::AXIsolatedObject::debugDescriptionInternal const):
(WebCore::AXIsolatedObject::dbgInternal const): Deleted.
* Source/WebCore/accessibility/isolatedtree/AXIsolatedObject.h:
* Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperBase.mm:
(-[WebAccessibilityObjectWrapperBase description]):
(-[WebAccessibilityObjectWrapperBase debugDescription]):

Canonical link: https://commits.webkit.org/297981@main
